core/src/main/java/org/mskcc/cbio/portal/servlet/CheckDarwinAccess.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2015 Memorial Sloan-Kettering Cancer Center.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF MERCHANTABILITY OR FITNESS
+ * FOR A PARTICULAR PURPOSE. The software and documentation provided hereunder
+ * is on an "as is" basis, and Memorial Sloan-Kettering Cancer Center has no
+ * obligations to provide maintenance, support, updates, enhancements or
+ * modifications. In no event shall Memorial Sloan-Kettering Cancer Center be
+ * liable to any party for direct, indirect, special, incidental or
+ * consequential damages, including lost profits, arising out of the use of this
+ * software and its documentation, even if Memorial Sloan-Kettering Cancer
+ * Center has been advised of the possibility of such damage.
+ */
+
+/*
+ * This file is part of cBioPortal.
+ *
+ * cBioPortal is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+package org.mskcc.cbio.portal.servlet;
+
+import org.mskcc.cbio.portal.model.DarwinAccess;
+
+import org.springframework.http.*;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.util.LinkedMultiValueMap;
+
+import org.apache.commons.cli.*;
+import org.mskcc.cbio.portal.util.GlobalProperties;
+import org.springframework.beans.factory.annotation.Value;
+
+/**
+ *
+ * @author ochoaa
+ */
+public class CheckDarwinAccess {
+    private static String DARWIN_AUTH_URL;
+    private static String DARWIN_RESPONSE_URL;
+    
+    public static String checkAccess(String userName, String patientId){
```

Comment:
We also want to take into account the cancer study - linking to darwin should only be relevant to mskimpact study/patients (we actually have many merged studies which contain mskimpact, but no easy way to determine that).  Also, we don't want to provide a link to darwin if the user came from the CIS system (see me to explain this).